pkg/controller/restore_controller.go

```diff
@@ -280,6 +292,11 @@ func (controller *restoreController) processRestore(key string) error {
 	for _, e := range restoreErrors.Namespaces {
 		restore.Status.Errors += len(e)
 	}
+	if restore.Status.Errors > 0 {
+		controller.metrics.RegisterRestoreFailed(backupScheduleName)
```

Comment:
I'm a bit on the fence about recording the restore as a "failure" if we have at least 1 error. It's possible that 99% of the restore completed ok, and some small portion had an error. What do you all think we should do here?

Reply:
@rosskukulinski @ashish-amarnath

Reply:
Would it maybe be more accurate to say that the restore was "incomplete"?

Reply:
+1, incomplete >>> failure imho

Reply:
What other reasons might a restore completely fail?

Reply:
- couldn't find the requested backup
- error downloading the backup tarball
- error creating temp file for the restore log
- error creating temp file for the restore warnings/errors
- error initializing restore item action plugins
- etc (there are more if you want me to keep listing)

Reply:
Looking at the possible phases for restores, we only have 4 - new, failed validation, in progress, or completed. So we don't have a specific phase that indicates complete failure. That may be useful to add.

For the metric, Incomplete makes sense.

Reply:
Yes, we should have Failed for complete failures, and we need to record the reason for the failure in restore.status.

Reply:
I like where this is going.  It's important to separate failures from incomplete.

Reply:
Metric name updated. The `restoreFailedTotal` metric can come back when the Failed phase is implemented.

Reply:
The more I think about it, the less useful I think this Incomplete metric is. It doesn't match an existing status, so the metric and the CRD will send a slightly different message. The information might be better captured by the error/warnings gauges, without a counter. Thoughts @ncdc, @rossfenrick @ashish-amarnath ?

Reply:
Want to add the `Failed` phase as part of this?

Reply:
@nrb: How/where are these gauges?  I don't think the fundamental behavior of gauges would work well with this type of data.

Reply:
@rosskukulinski They're hypothetical at the moment. I'm going to take a stab at implementing a Failed phase and using that for the Failed metric and defer reporting the number of errors/warnings to a later PR.